comp/forwarder/defaultforwarder/worker.go

```diff
@@ -84,7 +90,22 @@ func NewWorker(
 
 // NewHTTPClient creates a new http.Client
 func NewHTTPClient(config config.Component) *http.Client {
-	transport := httputils.CreateHTTPTransport(config)
+	var transport *http.Transport
+
+	transportConfig := config.Get("forwarder_http_protocol")
+
+	switch transportConfig {
+	case "http1":
+		transport = httputils.CreateHTTPTransport(config, func(t *http.Transport) {
+			t.MaxConnsPerHost = 1
+		})
+	case "auto":
+		fallthrough
+	default:
+		transport = httputils.CreateHTTPTransport(config, httputils.WithHTTP2(), func(t *http.Transport) {
```

Comment:
`auto` seems to mean "always http2", or am I mistaken? In a environment with pre-existing proxy configuration this is unlikely to work, but `auto` seems to imply we detect available protocol. 

Since there are only two options, would it make sense to use a boolean parameter instead? If not, we should probably warn about unexpected values rather than silently ignoring them.

Reply:
`auto` - or indeed anything other than "http1" means the agent will attempt HTTP/2 but continue with HTTP/1 if the connection doesn't support it.

I have copied these options from the logs config to stay consistent, so not sure if I should change..

Reply:
I've added a warn, but the diagnose package calls this too and from what I can see this doesn't have a log component to pass, so I have to check it isn't nil before warning. Not happy about this but can't think of a better way..